Query tests

alxsabo opened the pull request:
Implement query functionality to client

alxsabo committed:
Query tests [WIP] (+24 squashed commits)

ml054 commented on src/Documents/Queries/IndexQueryBase.php:
```diff
@@ -0,0 +1,167 @@
+        return $$this->query;
+    }
+
+    public function equals(?object &$o): bool {
+        if ($this == $o) return true;
+        if (($o == null) || (get_class($this) != get_class($o))) return false;
+
+        /** @var IndexQueryBase $that */
+        $that = $o;
+
+        if ($this->pageSize != $that->pageSize) return false;
+        if ($this->pageSizeSet != $that->pageSizeSet) return false;
+        if ($this->start != $that->start) return false;
+        if ($this->waitForNonStaleResults != $that->waitForNonStaleResults) return false;
+        if (($this->query != null) ? (strcmp($this->query, $that->query)) : ($that->query != null)) return false;
+        return $this->waitForNonStaleResultsTimeout != null ? $this->waitForNonStaleResultsTimeout->equals($that->waitForNonStaleResultsTimeout) : $that->waitForNonStaleResultsTimeout == null;
+    }
+
+    public function hashCode(): int
+    {
+        $result = $this->pageSize;
+        $result = 31 * $result + ($this->pageSizeSet ? 1 : 0);
+        $result = 31 * $result + ($this->query != null ? HashUtils::hashCode($this->query) : 0);
+        $result = 31 * $result + $this->start;
+        $result = 31 * $result + ($this->waitForNonStaleResults ? 1 : 0);
+        $result = 31 * $result + ($this->waitForNonStaleResultsTimeout != null ? $this->waitForNonStaleResultsTimeout->hashCode() : 0);
+        return $result;
+    }
```
usually you don't have to rewrite equals and hashcode functions. Moreover those are autogenerated in java. This is used for having such objects as map keys to compute hash backup and compare objects based on contents not via reference.

alxsabo replied:
at the beginning i didn't understood this methods so as you can see I just copied them. Now I understand it, and I will remove this methods soon

alxsabo committed:
Query tests (+4 squashed commits)

alxsabo committed:
Query tests (+19 squashed commits)

alxsabo committed:
disable non working test

alxsabo committed:
Query tests (+6 squashed commits)

alxsabo committed:
Query tests (+28 squashed commits)

alxsabo committed:
Query tests (+4 squashed commits)

ml054 committed:
added php 8.0 and 8.1 to build matrix + using separate dir for secured server data

alxsabo committed:
Query tests (+1 squashed commit)
Squashed commits:
[88c6dfe] Query tests (+2 squashed commits)
Squashed commits:
[6b75e00] Query tests
[335aebc] Query tests

alxsabo committed:
Merge pull request #1 from ml054/v5.2

added php 8.0 and 8.1 to build matrix + using separate dir for secure…

alxsabo committed:
Query tests

alxsabo committed:
Query tests (+2 squashed commits)

ml054 commented:
QueryTests duration fails due to invalid date format. You have:
![image](https://user-images.githubusercontent.com/1250431/165371901-0b6c8f48-64e8-4ebb-96f6-32f48e6685d6.png)
(4 digits after dot)

So please save with 3 digits or with 7 digits (in Java I fill this with four zeros, see: https://github.com/ravendb/ravendb-jvm-client/blob/d4e3dc2f60d5ba15f50136bf269e689526ae2fde/src/main/java/net/ravendb/client/primitives/NetISO8601Utils.java#L73)


Also this place might be needed: https://github.com/alxsabo/ravendb-php-client/blob/95fe801330c22ab8fe310ae838007ce2f98f17c0/src/Documents/Session/AbstractDocumentQuery.php#L1634

alxsabo committed:
Fix date time normalizer

alxsabo committed:
Query tests
